wazo_webhookd/plugins/mobile/celery_tasks.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2019-2023 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Literal, overload
+
+import requests
+from celery import Task
+from pkg_resources import EntryPoint
+
+from wazo_webhookd.celery import app
+from .schema import NotificationDict
+
+if TYPE_CHECKING:
+    from ...services.mobile.plugin import (
+        PushNotification,
+        Service as PushNotificationService,
+    )
+    from ...types import WebhookdConfigDict
+
+
+MOBILE_SERVICE_ENTRYPOINT = 'mobile = wazo_webhookd.services.mobile.plugin'
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+
+    @overload
+    def get_service_class(name: Literal['Service']) -> type[PushNotificationService]:
+        ...
+
+    @overload
+    def get_service_class(name: Literal['PushNotification']) -> type[PushNotification]:
+        ...
+
+
+def get_service_class(name: str) -> type[PushNotificationService | PushNotification]:
+    return EntryPoint.parse(f'{MOBILE_SERVICE_ENTRYPOINT}:{name}').resolve()
+
+
+@app.task(bind=True)
+def send_notification(
+    task: Task,
+    config: WebhookdConfigDict,
+    notification: NotificationDict,
+) -> bool:
+    service_class: type[PushNotificationService] = get_service_class('Service')
+    notification_class: type[PushNotification] = get_service_class('PushNotification')
```

Comment:
Why use the entrypoint instead of importing the class directly? Is it because the class comes from a Python plugin?

I think we should extract this code to the webhookd core anyway...

Reply:
I thought that was the objective? I was doing what was done elsewhere. These are plugins and if it's disable the import will fail. I've never really understood the way we use plugins in a lot of cases. A ton of code is in plugins, but a lot of the time nothing will work if they're disable and we just expect most of them to be there.

Reply:
Agreed, we don't have clear specs on how to depend on plugins. I just don't remember having seen this pattern elsewhere in Wazo code. Anyway, no objection to do it this way.

Reply:
Well I find direct imports to be simpler to read, even though it tightens the coupling between "plugins". No objection to keep it like this, your call :)

Reply:
Oh yeah, for sure. Direct imports are definitely easier. I only did this to keep the notion of plugins. I'm all for simplifying.